pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -120,6 +127,10 @@ def __init__(
             )
         self.monitor = monitor or "early_stop_on"
 
+    @property
+    def state_key(self) -> str:
+        return self._generate_state_key(monitor=self.monitor, mode=self.mode)
+
```

Comment:
nitpicky: we could split out the implementations of these `state_key`s in various callbacks to a separate PR

Reply:
there are only these two callbacks right now that need this special treatment. no other callbacks need a state_key right now afaik.